eftcamb/11_EFTCAMB_stability.f90

```diff
@@ -33,211 +33,119 @@ module EFTCAMB_stability
-!        end select
-!        ! compute all other background stuff:
-!        call CP%EFTCAMB%model%compute_rhoQPQ( a, CP%eft_par_cache , eft_cache )
-!        ! compute second order EFT functions:
-!        call CP%EFTCAMB%model%compute_secondorder_EFT_functions( a, CP%eft_par_cache , eft_cache )
+       ! Stability check initialization
+       EFTStabilityComputation = .true.
+       ! prepare:
+       a2 = a*a
+       ! initialize the cache:
+       call eft_cache%initialize()
+       ! start filling:
+       eft_cache%a = a
+       ! compute background densities of different species
+       eft_cache%grhob_t = eft_par_cache%grhob/a         ! 8\pi G_N \rho_b a^2: baryons background density
+       eft_cache%grhoc_t = eft_par_cache%grhoc/a         ! 8\pi G_N \rho_{cdm} a^2: cold dark matter background density
+       eft_cache%grhor_t = eft_par_cache%grhornomass/a2  ! 8\pi G_N \rho_{\nu} a^2: massless neutrinos background density
+       eft_cache%grhog_t = eft_par_cache%grhog/a2        ! 8\pi G_N \rho_{\gamma} a^2: radiation background density
+       ! Massive neutrinos terms:
+       if ( CP%Num_Nu_Massive /= 0 ) then
+           do nu_i = 1, CP%Nu_mass_eigenstates
+               EFT_grhonu    = 0._dl
+               EFT_gpinu     = 0._dl
+               grhormass_t=grhormass(nu_i)/a**2
+               call Nu_background(a*nu_masses(nu_i),EFT_grhonu,EFT_gpinu)
+               eft_cache%grhonu_tot = eft_cache%grhonu_tot + grhormass_t*EFT_grhonu
+               eft_cache%gpinu_tot  = eft_cache%gpinu_tot  + grhormass_t*EFT_gpinu
+           end do
+       end if
+       ! assemble total densities and pressure:
+       eft_cache%grhom_t  = eft_cache%grhob_t +eft_cache%grhoc_t +eft_cache%grhor_t +eft_cache%grhog_t +eft_cache%grhonu_tot
+       eft_cache%gpresm_t = (+eft_cache%grhor_t +eft_cache%grhog_t)/3._dl +eft_cache%gpinu_tot
+       ! compute the other things:
+       select type ( model => CP%EFTCAMB%model )
+           ! compute the background and the background EFT functions.
+           class is ( EFTCAMB_full_model )
+           ! background for full models. Here the expansion history is computed from the
+           ! EFT functions. Hence compute them first and then compute the expansion history.
+           call CP%EFTCAMB%model%compute_background_EFT_functions( a, CP%eft_par_cache , eft_cache )
+           call CP%EFTCAMB%model%compute_adotoa( a, CP%eft_par_cache , eft_cache )
+           class is ( EFTCAMB_designer_model )
+           ! background for designer models. Here the expansion history is parametrized
+           ! and does not depend on the EFT functions. Hence compute first the expansion history
+           ! and then the EFT functions.
+           call CP%EFTCAMB%model%compute_adotoa( a, CP%eft_par_cache , eft_cache )
+       end select
+       ! store adotoa:
+       adotoa   = eft_cache%adotoa  !SP: is this really needed?
```

Comment:
This is used in a couple of places below. Fine for me if you want to remove it and make it consistent later!